packages/suite/src/components/suite/SecurityCheck/useReportDeviceCompromised.ts

```diff
@@ -26,34 +30,29 @@ const useCommonData = () => {
 };
 
 const useReportRevisionCheck = () => {
-    const { device } = useDevice();
     const commonData = useCommonData();
-
-    const revisionCheck = isDeviceAcquired(device)
-        ? device.authenticityChecks?.firmwareRevision
-        : null;
-    const isRevisionCheckError = revisionCheck && !revisionCheck.success;
-    const revisionCheckError = isRevisionCheckError ? revisionCheck.error : null;
+    const revisionCheckError = useSelector(selectFirmwareRevisionCheckError);
 
     useEffect(() => {
-        if (!isRevisionCheckError) return;
-        reportCheckFail('revision', { ...commonData, revisionCheckError });
-    }, [commonData, isRevisionCheckError, revisionCheckError]);
+        if (revisionCheckError !== null) {
+            reportCheckFail('revision', { ...commonData, revisionCheckError });
+        }
+    }, [commonData, revisionCheckError]);
 };
 
 const useReportHashCheck = () => {
     const { device } = useDevice();
     const commonData = useCommonData();
 
+    const hashCheckError = useSelector(selectFirmwareHashCheckError);
     const hashCheck = isDeviceAcquired(device) ? device.authenticityChecks?.firmwareHash : null;
-    const isHashCheckError = hashCheck && !hashCheck.success;
-    const hashCheckError = isHashCheckError ? hashCheck.error : null;
-    const hashCheckErrorPayload = isHashCheckError ? hashCheck.errorPayload : null;
+    const hashCheckErrorPayload = hashCheck?.success === false ? hashCheck.errorPayload : null;
 
     useEffect(() => {
-        if (!isHashCheckError) return;
-        reportCheckFail('hash', { ...commonData, hashCheckError, hashCheckErrorPayload });
-    }, [commonData, hashCheckError, isHashCheckError, hashCheckErrorPayload]);
+        if (hashCheckError !== null) {
+            reportCheckFail('hash', { ...commonData, hashCheckError, hashCheckErrorPayload });
+        }
+    }, [commonData, hashCheckError, hashCheckErrorPayload]);
```

Comment:
`useReportRevisionCheck` is now much nicer thanks to the new selector that gets error regardless of `enabled`.
But `useReportHashCheck` also needs payload so the code is not simplified. But at least common logic is deduplicated!

:thought_balloon: I could create `selectHashCheckErrorPayload` but sounds like overkill..